service/kvmclusterconfig/framework.go

```diff
@@ -11,16 +11,18 @@ import (
 	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
 	"k8s.io/client-go/kubernetes"
 
+	"github.com/giantswarm/cluster-operator/pkg/cluster"
 	"github.com/giantswarm/cluster-operator/service/kvmclusterconfig/v1"
 )
 
 // FrameworkConfig contains necessary dependencies and settings for
 // KVMClusterConfig CRD framework implementation.
 type FrameworkConfig struct {
-	G8sClient    versioned.Interface
-	K8sClient    kubernetes.Interface
-	K8sExtClient apiextensionsclient.Interface
-	Logger       micrologger.Logger
+	BaseClusterConfig *cluster.Config
```

Comment:
I still don't get what this is doing here, because it is not used. Is this preparation for next steps? If so, I would have appreciated some information upfront then I could have saved some time trying to understand what is going on here and why.

Reply:
Yes, this is preparation here for `certconfig` resource to be added to resource set. I also created this here already so that I could guarantee that my idea with configuration parsing/passing works e2e. Apologies for distraction. I thought this would be clear from context within this PR.